network.go

```diff
@@ -10,3 +10,45 @@ func parseHeaders(headers []interface{}) map[string]string {
 	}
 	return out
 }
+
+type rawHeaders struct {
+	headersArray map[string]string
+	headersMap   map[string][]string
+}
+
+func (r *rawHeaders) Get(name string) string {
+	values := r.GetAll(name)
+	sep := ", "
+	if strings.ToLower(name) == "set-cookie" {
+		sep = "; "
+	}
+	return strings.Join(values, sep)
+}
+func (r *rawHeaders) GetAll(name string) []string {
+	return r.headersMap[strings.ToLower(name)]
+}
+func (r *rawHeaders) Headers() map[string]string {
+	out := make(map[string]string)
+	for key, value := range r.headersArray {
```

Comment:
this should loop over the keys of headersMap and use the r.Get method instead.

Reply:
In a follow up will fix all network related